pkg/demo/demo.go

```diff
@@ -85,12 +85,28 @@ func New(pgpwd string) (*Demo, error) {
 	return app, nil
 }
 
+func (app *Demo) Query(ctx context.Context, dataset string, rows, cols []string) (string, error) {
+	if dataset == "skinny" {
+		return app.skinnyQuery(ctx, rows, cols)
+	}
+	cols = gatherTokens(cols)
+	return app.tableQuery(ctx, dataset, rows, cols)
+}
+
+func (app *Demo) skinnyQuery(ctx context.Context, rows, cols []string) (string, error) {
+	clause, err := where.SkinnyWhere(rows, cols)
+	if err != nil {
+		return "", err
+	}
+	return clause, nil
+}
+
 // query runs a SQL query against the db and returns the resulting CSV as a string.
 // dataset is the name of the table to query.
 // rowspec is not used at present.
 // colspec is a list of columns to include in the result. Empty means all columns.
 //
-func (app *Demo) Query(ctx context.Context, dataset string, rowspec, colspec []string) (string, error) {
+func (app *Demo) tableQuery(ctx context.Context, dataset string, rowspec, colspec []string) (string, error) {
```

Comment:
Realise its WIP, but the skinnyQuery / tableQuery naming seems a little off to me. Can't immediately think of better options though.

Reply:
yep, open to suggestions